frontend/src/components/generalButtons/button.tsx

```diff
@@ -0,0 +1,67 @@
+import PropTypes from 'prop-types';
+import {Button, IconButton, Tooltip} from '@mui/material';
+import {CustomButtonProps} from '../../types';
+
+const CustomButton: React.FC<CustomButtonProps> = props => {
+  const renderIcon = () => {
+    return props.icon && <IconButton>props.icon</IconButton>;
+  };
+
+  const renderBtnContent = () => {
+    return (
+      <>
+        {props.label}
+        {renderIcon()}
+      </>
+    );
+  };
+  const button_component = (
+    <Button
+      variant={props.variant}
+      color={props.color}
+      size={props.size}
+      disabled={props.disabled}
+      href={props.href ? props.href : '#'}
```

Comment:
Using href here will reload the page. Instead of this use Link element from `react-router-dom`.